test/TritonGPU/amd/amd-schedule-hint.mlir

```diff
@@ -0,0 +1,74 @@
+// RUN: triton-opt %s -split-input-file -triton-amdgpu-insert-instruction-sched-hints='variant=attention' | FileCheck %s -check-prefix=INSTR_HINT
+// RUN: triton-opt %s -split-input-file -triton-amdgpu-insert-instruction-sched-hints='variant=attention' -triton-amdgpu-lower-insert-instruction-sched-hints -verify-diagnostics | FileCheck %s -check-prefix=LOWER_HINT
+
+#blocked = #ttg.blocked<{sizePerThread = [4, 4], threadsPerWarp = [8, 8], warpsPerCTA = [2, 4], order = [1, 0]}>
+#mma = #ttg.amd_mfma<{versionMajor = 3, versionMinor = 0, warpsPerCTA = [2, 4], instrShape = [32, 32], isTransposed = true}>
+#dot_op_a = #ttg.dot_op<{opIdx = 0, parent = #mma, kWidth = 4}>
+#dot_op_b = #ttg.dot_op<{opIdx = 1, parent = #mma, kWidth = 4}>
+// INSTR_HINT-LABEL: @insert_schedule_hint
+module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 8 : i32, ttg.target = "hip:gfx942", "ttg.threads-per-warp" = 64 : i32} {
+  tt.func public @insert_schedule_hint(
+    %lb : index, %ub : index, %step : index,
+    %arg0: tensor<128x128xf32, #dot_op_a>,
+    %arg1: tensor<128x128xf32, #dot_op_b>,
+    %arg2: tensor<128x128x!tt.ptr<f32>, #blocked>
+  ) {
+    %cst = arith.constant dense<0.000000e+00> : tensor<128x128xf32, #mma>
+    // INSTR_HINT: scf.for
+    // INSTR_HINT-NEXT: amdgpu.instruction_sched_hint
+    // INSTR_HINT-SAME: isBufferLoadsAEnabled = false
+    // INSTR_HINT-SAME: isBufferLoadsBEnabled = false
+    // INSTR_HINT-SAME: numDsReadsA = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numDsReadsB = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numDsWritesA = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numDsWritesB = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numGlobalLoadsA = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numGlobalLoadsB = #amdgpu.InstCounter<0, none>
+    // INSTR_HINT-SAME: numMMAs = #amdgpu.InstCounter<0, none>
```

Comment:
These are irrelevant here right? Don't think we need to check these?

Reply:
Ah yeah actually only the last line matters.